src/test/client.rs

```diff
@@ -180,13 +181,72 @@ async fn list_database_names() {
             .unwrap();
     }
 
-    let new_dbs = client.list_database_names(None).await.unwrap();
+    let new_dbs = client.list_database_names(None, None).await.unwrap();
 
     for name in expected_dbs {
         assert_eq!(new_dbs.iter().filter(|db_name| db_name == &name).count(), 1);
     }
 }
 
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn list_authorized_databases() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || !client.auth_enabled() {
+        return;
+    }
+
+    let dbs = &[
+        format!("{}1", function_name!()),
+        format!("{}2", function_name!()),
+    ];
+
+    for name in dbs {
+        client
+            .database(name)
+            .create_collection("coll", None)
+            .await
+            .unwrap();
+        client
+            .create_user(
+                &format!("user_{}", name),
+                "pwd",
+                &[Bson::from(doc! { "role": "readWrite", "db": name })],
+                &[AuthMechanism::ScramSha256],
+            )
+            .await
+            .unwrap();
+    }
+
+    for name in dbs {
+        let mut options = CLIENT_OPTIONS.clone();
+        let credential = Credential::builder()
+            .username(Some(format!("user_{}", name)))
```

Comment:
Since the builder methods take `impl Into<Option<T>>` rather than just `Option<T>`, you can leave off the `Some(...)` and just put the value directly, e.g. `.username(format!("user_{}", name))`. This is because `Option<T>` implements `From<T>`.